eth/eventservices/jobservice.go

```diff
@@ -46,8 +49,17 @@ func (s *JobService) Start(ctx context.Context) error {
 	sub, err := s.eventMonitor.SubscribeNewJob(ctx, "NewJob", logsCh, common.Address{}, func(l types.Log) (bool, error) {
 		_, jid, _, _ := parseNewJobLog(l)
 
-		job, err := s.node.Eth.GetJob(jid)
-		if err != nil {
+		var job *lpTypes.Job
+		getJob := func() error {
+			j, err := s.node.Eth.GetJob(jid)
+			if j.StreamId == "" {
```

Comment:
Hm so when using Infura, an event can come back with empty fields? Weird. I think there is an edge case where someone actually creates a job with an empty streamID, but I guess in that scenario, after a number of retries the transcoder will give up and just drop the new job event

Reply:
Yeah it's really strange... But I've seen that a few times that crashed the transcoder. https://github.com/livepeer/go-livepeer/issues/342